Sources/SecureXPC/Server/XPCServer.swift

```diff
@@ -291,7 +291,7 @@ public class XPCServer {
         // Keep a weak reference to the connection and this server, setting this as the context on the connection
         let weakConnection = WeakConnection(connection, server: self)
         self.connections.insert(weakConnection)
-        xpc_connection_set_context(connection, Unmanaged.passUnretained(weakConnection).toOpaque())
+        xpc_connection_set_context(connection, Unmanaged.passRetained(weakConnection).toOpaque())
```

Comment:
Hmmm I didn't think this was necessary, because I thought that the `connections` array would be holding a strong reference.

In retrospect, that was silly, I was acting as if strong refs should be spared. I think passing retained, and taking retained is the correct semantics, but only if the finalizer below is guaranteed to be called,  exactly once.

Reply:
I'm pretty confident the finalizer can only be called once because the documentation says:
> The function that will be invoked when the connection's retain count has dropped to zero and is being torn down.